nix/dev-shell.nix

```diff
@@ -0,0 +1,25 @@
+{ cargo
+, clippy
+, elmPackages
+, mkShell
+, openssl
+, pkg-config
+, rustc
+, rustfmt
+}:
```

Comment:
Can this be formatted a little more sane way? adding fields would produce awful diffs

Reply:
Would it make sense to add a formatter to the flake? So that we can run `nix fmt`?

Reply:
yea, probably

Reply:
added `nixpkgs-fmt` which ironically didn't format this file.

I'm not personally a fan of the nix rfc formatter

Reply:
My ideal would be https://github.com/kamadorueda/alejandra if we could have surrounding whitespace: `{mkDerviation}:` -> `{ mkDerivation }:`

Reply:
>  but generally this formatting is... questionable

This was considered best practice until the nix formatting rfc